presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -257,14 +256,10 @@ public OrcRecordReader(
                 writeValidation);
 
         streamReaders = createStreamReaders(orcDataSource, types, hiveStorageTimeZone, presentColumnsAndTypes.build(), streamReadersSystemMemoryContext);
-        if (includedColumnHandles != null) {
-            for (int columnIdx = 0; columnIdx < streamReaders.length; columnIdx++) {
-                ColumnHandle columnHandle = includedColumnHandles.get(columnIdx);
-                if (columnHandle != null && streamReaders[columnIdx] != null && columnHandle.getReferencedSubfields() != null) {
-                    streamReaders[columnIdx].setReferencedSubfields(columnHandle.getReferencedSubfields(), 0);
-                }
-            }
-        }
+        includedColumnHandles.entrySet().stream()
+                .filter(e -> e.getValue().getReferencedSubfields() != null)
+                .filter(e -> streamReaders[e.getKey()] != null)
+                .forEach(e -> streamReaders[e.getKey()].setReferencedSubfields(e.getValue().getReferencedSubfields(), 0));
```

Comment:
I usually see the stream api used for simple statements, it seems easier to reason about the for loops for more complex logic. This is definitely just an opinion though.